python/pyspark/pandas/tests/data_type_ops/test_categorical_ops.py

```diff
@@ -183,22 +183,31 @@ def test_astype(self):
         psser = ps.from_pandas(pser)
         self.assert_eq(pser.astype(int), psser.astype(int))
         self.assert_eq(pser.astype(float), psser.astype(float))
-        self.assert_eq(pser.astype(np.float32), psser.astype(np.float32))
-        self.assert_eq(pser.astype(np.int32), psser.astype(np.int32))
-        self.assert_eq(pser.astype(np.int16), psser.astype(np.int16))
-        self.assert_eq(pser.astype(np.int8), psser.astype(np.int8))
         self.assert_eq(pser.astype(str), psser.astype(str))
-        self.assert_eq(pser.astype(bool), psser.astype(bool))
         self.assert_eq(pser.astype("category"), psser.astype("category"))
 
+        # Bug in pandas 1.2
+        if (LooseVersion(pd.__version__) >= LooseVersion("1.2")) and (
+            LooseVersion(pd.__version__) < LooseVersion("1.3")
+        ):
+            self.assert_eq(psser.astype(np.float32), ps.Series(data, dtype=np.float32))
+            self.assert_eq(psser.astype(np.int32), ps.Series(data, dtype=np.int32))
+            self.assert_eq(psser.astype(np.int16), ps.Series(data, dtype=np.int16))
+            self.assert_eq(psser.astype(np.int8), ps.Series(data, dtype=np.int8))
+            self.assert_eq(psser.astype(bool), ps.Series(data, dtype=bool))
+        else:
+            self.assert_eq(pser.astype(np.float32), psser.astype(np.float32))
+            self.assert_eq(pser.astype(np.int32), psser.astype(np.int32))
+            self.assert_eq(pser.astype(np.int16), psser.astype(np.int16))
+            self.assert_eq(pser.astype(np.int8), psser.astype(np.int8))
+            self.assert_eq(pser.astype(bool), psser.astype(bool))
```

Comment:
Do we need this? It seems working with pandas 1.2 without this change.

Reply:
Oh... then seems like we're missing tests for pandas 1.2 ??

It's failed on my local with pandas 1.2.

```
Attribute "dtype" are different
[left]:  float64
[right]: float32

Left:
0    1.0
1    2.0
2    3.0
dtype: float64
float64

Right:
0    1.0
1    2.0
2    3.0
dtype: float32
float32
```

The result is actually different.

```python
>>> pd.__version__
'1.2.0'
>>> pser.astype(np.float32)
0    1.0
1    2.0
2    3.0
dtype: float64
>>> psser.astype(np.float32)
0    1.0
1    2.0
2    3.0
dtype: float32
```

Reply:
Seems like it's a pandas' bug?
`pser.astype(np.float32)` should definitely return `dtype: float32`..
Could you narrow down the version range then?

Reply:
pandas 1.2.5 seems working, btw.

Reply:
@ueshin Oh, yes I'll narrow down the version range after checking the all 1.2.x versions.

Thanks!